web/profiles/custom/os2loop/modules/os2loop_toc_block/src/Plugin/Block/TocBlock.php

```diff
@@ -0,0 +1,159 @@
+
+  /**
+   * The toc builder interface.
+   *
+   * @var \Drupal\toc_api\TocBuilderInterface
+   */
+  protected $tocBuilder;
+
+  /**
+   * The route match interface.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Drupal\Core\Render\RendererInterface definition.
+   *
+   * @var \Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * Block constructor for table of contents.
+   *
+   * @param array $configuration
+   *   The block configuration.
+   * @param string $plugin_id
+   *   The block id.
+   * @param mixed $plugin_definition
+   *   The block definition.
+   * @param \Drupal\toc_api\TocManagerInterface $tocManager
+   *   Manager service for table of contents.
+   * @param \Drupal\toc_api\TocBuilder $tocBuilder
+   *   Builder service for table of contents.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $routeMatch
+   *   Route match service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   Drupals renderer service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, TocManagerInterface $tocManager, TocBuilder $tocBuilder, RouteMatchInterface $routeMatch, EntityTypeManagerInterface $entityTypeManager, RendererInterface $renderer) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->tocManager = $tocManager;
+    $this->tocBuilder = $tocBuilder;
+    $this->routeMatch = $routeMatch;
+    $this->entityTypeManager = $entityTypeManager;
+    $this->renderer = $renderer;
+  }
+
+  /**
+   * Create method for block.
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The service container.
+   * @param array $configuration
+   *   The block configuration.
+   * @param string $plugin_id
+   *   The block id.
+   * @param mixed $plugin_definition
+   *   The block definition.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('toc_api.manager'),
+      $container->get('toc_api.builder'),
+      $container->get('current_route_match'),
+      $container->get('entity_type.manager'),
+      $container->get('renderer'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $lock = &drupal_static(__FUNCTION__, FALSE);
+    $build = [];
+
+    if (!$lock) {
+      $lock = TRUE;
+      $node = $this->routeMatch->getParameter('node');
+      if (!($node instanceof NodeInterface)) {
+        return $build;
+      }
```

Comment:
This will not work when generating content for PDF export, i.e. when the `entity_print.view` route is used (run `vendor/bin/drupal debug:router entity_print.view` to see route details).

Is this by design or by mistake (that we don't generate a TOC in a PDF export)?